pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -358,6 +358,12 @@ func (h *handler) createNewCluster(cluster *v1.Cluster, status v1.ClusterStatus,
 		Spec: spec,
 	}
 
+	if newCluster.Spec.Rke2Config == nil {
```

Comment:
I'd suggest not adding this because the `newCluster` could be a K3s cluster.

Reply:
I removed it.